Classes/Extensions/DDFileLogger+Buffering.m

```diff
@@ -0,0 +1,165 @@
+// Software License Agreement (BSD License)
+//
+// Copyright (c) 2010-2018, Deusty, LLC
+// All rights reserved.
+//
+// Redistribution and use of this software in source and binary forms,
+// with or without modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+#import "DDFileLogger+Buffering.h"
+
+@interface DDFileLogger (Internal)
+- (void)logData:(NSData *)data;
+@end
+
+// TODO: Fix it later. ( Protocol adoption? )
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wincomplete-implementation"
+@implementation DDFileLogger (Internal)
+@end
+#pragma clang diagnostic pop
+
+static NSUInteger kMaximumBytesCountInBuffer = (1 << 10) * (1 << 10); // 1 MB.
+static NSUInteger kDefaultBytesCountInBuffer = (1 << 10);
+
+// MARK: Public Interface
+@interface DDBufferedProxy<T: DDFileLogger */*id<DDLogger>*/>: NSProxy
```

Comment:
Nice idea to make it generic. But since it's not a public proxy (nor should it be right now), I don't think that we should add the complexity of making it generic for now.

Reply:
If we keep it, we should do it right, at least:
- `T` should be named `FileLogger`
- `/*id<DDLogger>*/` can be removed to make it more readable.
- In the implementation, we should then also use the generic type and not `DDFileLogger`.

Reply:
AFAIK they're raised to the highest possible type. In our case `DDFileLogger`. So the `typedef` is not necessary. Since we don't use it with a subclass of `DDFileLogger`, using `__kindof` will not change anything. Maybe even complicate compilation.